### PR TITLE
Add proxy support

### PR DESCRIPTION
**Pull request by JulienDev**

### CODE REVIEW

File: instapy/instapy.py [hunk 1]
```diff
@@ -149,6 +153,13 @@ def set_selenium_local_session(self):
             # this setting can improve pageload & save bandwidth
             firefox_profile.set_preference('permissions.default.image', 2)
 
+            if self.proxy_address is not None and self.proxy_port > 0:
```
Comment (timgrossmann) [inline instapy/instapy.py]: Could we do:
`if self.proxy_address and self.proxy_port > 0:`
?

File: instapy/instapy.py [hunk 2]
```diff
@@ -158,7 +169,11 @@ def set_selenium_local_session(self):
             chrome_options.add_argument('--no-sandbox')
             chrome_options.add_argument('--lang=en-US')
             chrome_options.add_argument('--disable-setuid-sandbox')
-            
+
+            if self.proxy_address is not None and self.proxy_port > 0:
```
Comment (timgrossmann) [inline instapy/instapy.py]: Could we do:
`if self.proxy_address and self.proxy_port > 0:`
?

File: instapy/instapy.py [hunk 3]
```diff
@@ -158,7 +169,11 @@ def set_selenium_local_session(self):
...
+                proxy = '%s:%s' % (self.proxy_address, self.proxy_port)
```
Comment (timgrossmann) [inline instapy/instapy.py]: Could we please use the same format as with the other strings?
`proxy = '{ip}:{port}'.format(ip=self.proxy_address, port=self.proxy_port)`
?

File: instapy/instapy.py [hunk 4]
```diff
@@ -158,7 +169,11 @@ def set_selenium_local_session(self):
...
+                chrome_options.add_argument('--proxy-server=http://%s' % proxy)
```
Comment (timgrossmann) [inline instapy/instapy.py]: Could we do:
`chrome_options.add_argument('--proxy-server=http://{proxy}'.format(proxy=proxy))`
?

**Review by timgrossmann — CHANGES_REQUESTED** [review]

**Comment by timgrossmann** [comment]
@JulienDev Thank you very much for this!
Could you make these small changes, please and we'll merge that after this.

So awesome, thanks for contributing!

**Comment by JulienDev** [comment]
Here you are! 
Sorry, it's my first time writing python.

**Review by timgrossmann — APPROVED** [review]

**Comment by timgrossmann** [comment]
@JulienDev No worries, I'll test it later today to make sure that it works :wink: 
Thank you for contributing!

**Comment by converge** [comment]
Sounds great ! We just need to fix the suspicious login attempt before merge it.

**Comment by sionking** [comment]
Working for me ! thanks.

**Comment by converge** [comment]
thanks @JulienDev !

**Comment by JulienDev** [comment]
You're welcome :)

Le 22 déc. 2017 17:31, "João Paulo Vanzuita" <notifications@github.com> a
écrit :

**Pull request merged**